src/MongoDB.Driver.Core/Core/Compression/Snappy/SnappyCodec.cs

```diff
@@ -0,0 +1,166 @@
+﻿/* Copyright 2019–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.IO;
+using MongoDB.Driver.Core.Misc;
+
+namespace MongoDB.Driver.Core.Compression.Snappy
+{
+    internal static class SnappyCodec
+    {
+        public static int Compress(byte[] input, int inputOffset, int inputLength, byte[] output, int outputOffset, int outputLength)
+        {
+            Ensure.IsNotNull(input, nameof(input));
+            Ensure.IsNotNull(output, nameof(output));
+            EnsureBufferRangeIsValid(inputOffset, inputLength, input.Length);
+            EnsureBufferRangeIsValid(outputOffset, outputLength, output.Length);
```

Comment:
Still not 100% sure that we can use the same validation method for `input` and `output`. It looks like it's true, but I want to make some more tests.